pallets/automation-time/src/tests.rs

```diff
@@ -217,75 +220,58 @@ fn schedule_not_enough_for_fees() {
 	})
 }
 
+// test schedule transfer with dynamic dispatch.
 #[test]
-fn schedule_native_transfer_invalid_amount() {
+fn schedule_transfer_with_dynamic_dispatch() {
 	new_test_ext(START_BLOCK_TIME).execute_with(|| {
-		assert_noop!(
-			AutomationTime::schedule_native_transfer_task(
-				RuntimeOrigin::signed(AccountId32::new(ALICE)),
-				vec![50],
-				vec![SCHEDULED_TIME],
-				AccountId32::new(BOB),
-				0,
-			),
-			Error::<Test>::InvalidAmount,
-		);
-	})
-}
+		let account_id = AccountId32::new(ALICE);
+		let provided_id = vec![1, 2];
+		let task_id = AutomationTime::generate_task_id(account_id.clone(), provided_id.clone());
+		let task_hash_input = TaskHashInput::new(account_id.clone(), vec![1, 2]);
 
-#[test]
-fn schedule_native_transfer_cannot_transfer_to_self() {
-	new_test_ext(START_BLOCK_TIME).execute_with(|| {
-		assert_noop!(
-			AutomationTime::schedule_native_transfer_task(
-				RuntimeOrigin::signed(AccountId32::new(ALICE)),
-				vec![50],
-				vec![SCHEDULED_TIME],
-				AccountId32::new(ALICE),
-				1,
-			),
-			Error::<Test>::TransferToSelf,
-		);
-	})
-}
+		fund_account(&account_id, 900_000_000, 2, Some(0));
 
-#[test]
-fn schedule_native_transfer_works() {
-	new_test_ext(START_BLOCK_TIME).execute_with(|| {
-		get_funds(AccountId32::new(ALICE));
-		assert_ok!(AutomationTime::schedule_native_transfer_task(
-			RuntimeOrigin::signed(AccountId32::new(ALICE)),
-			vec![50],
-			vec![SCHEDULED_TIME],
-			AccountId32::new(BOB),
-			1,
+		let call: <Test as frame_system::Config>::RuntimeCall =
+			pallet_balances::Call::transfer { dest: AccountId32::new(BOB), value: 127 }.into();
+
+		assert_ok!(AutomationTime::schedule_dynamic_dispatch_task(
+			RuntimeOrigin::signed(account_id.clone()),
+			vec![1, 2],
+			ScheduleParam::Fixed { execution_times: vec![SCHEDULED_TIME] },
+			Box::new(call),
 		));
-		match AutomationTime::get_scheduled_tasks(SCHEDULED_TIME) {
-			None => {
-				panic!("A task should be scheduled")
-			},
-			Some(ScheduledTasksOf::<Test> { tasks: account_task_ids, .. }) =>
-				match AutomationTime::get_account_task(
-					account_task_ids[0].0.clone(),
-					account_task_ids[0].1,
-				) {
-					None => {
-						panic!("A task should exist if it was scheduled")
-					},
-					Some(task) => {
-						let expected_task = TaskOf::<Test>::create_native_transfer_task::<Test>(
-							AccountId32::new(ALICE),
-							vec![50],
-							vec![SCHEDULED_TIME],
-							AccountId32::new(BOB),
-							1,
-						)
-						.unwrap();
-
-						assert_eq!(task, expected_task);
-					},
-				},
-		}
+
+		Timestamp::set_timestamp(SCHEDULED_TIME * 1_000);
+		LastTimeSlot::<Test>::put((LAST_BLOCK_TIME, LAST_BLOCK_TIME));
+		System::reset_events();
+
+		let w = AutomationTime::trigger_tasks(Weight::from_ref_time(900_000_000));
+		let my_events = events();
+
+		let bob = AccountId32::new(BOB);
```

Comment:
Let’s not use name such as bob for a wallet, instead use a functional name. For example, if bob is the creator of the task, call it `creator` or `signer`.

This rule should apply to all tests, and we can update them whenever we work on other tests. Let me know what you think.